js/main.js

```diff
@@ -0,0 +1,144 @@
+ * own square within the board.
+ * @param {number} canvasHeight
+ * @param {number} canvasWidth
+ * @returns {Object}
+ */
+function createGameBoard(canvasHeight, canvasWidth) {
+  let gameBoard =
+      {startPointX: 0, startPointY: 0, gameHeight: 10, gameWidth: 10, board: []}
+
+  let heightScaler = canvasHeight / gameBoard.gameHeight;
+  let widthScaler = canvasWidth / gameBoard.gameWidth;
+
+  if (widthScaler > heightScaler) {
+    widthScaler = heightScaler;
+  } else {
+    heightScaler = widthScaler;
+  }
+
+  let middlePoint = canvasWidth / 2 - (widthScaler * gameBoard.gameWidth / 2);
+  gameBoard.startPointX = middlePoint;
+
+  currX = gameBoard.startPointX;
+  currY = gameBoard.startPointY;
+
+  for (let i = 0; i < gameBoard.gameHeight; ++i) {
+    let rowOfSquares = [];
+    for (let j = 0; j < gameBoard.gameWidth; ++j) {
+      let square = {
+        xCoord: currX,
+        yCoord: currY,
+        width: widthScaler,
+        height: heightScaler,
+        character: '-'
+      };
+
+      rowOfSquares.push(square);
+
+      currX += widthScaler;
+    }
+    gameBoard.board.push(rowOfSquares);
+    currX = gameBoard.startPointX;
+    currY += heightScaler;
+  }
+
+  return gameBoard;
+}
+
+/**
+ * Converts ranNumber from a float to a character in the
+ * english alphabet based on what interval it is within the charFrequency.
+ * @param {array} charFrequency an array of floats where element at index i
+ *    less element index i - 1 is the frequency that the i'th letter occurs in
+ *    the english language
+ * @param {number} ranNumber a random float between 0-101.432
+ * @return {String}
+ */
+function getCharater(charFrequency, ranNumber) {
+  let index = 0;
+
+  for (let i = 0; i < 26; ++i) {
+    if (ranNumber < charFrequency[i]) {
+      index = i;
+      break;
+    }
+  }
+
+  return String.fromCharCode(97 + index)
+}
+
+/**
+ * Based on a weighted random distribution, it populates the wordSearchBoard
+ * with random letters.
+ * @param {object} wordSearchBoard
+ */
+function randomizeCharacters(wordSearchBoard) {
+  /** @const {number} */
+  const frequency = [
+    8.498,  9.989,  12.191, 16.445, 27.607, 29.835, 31.85,   37.944, 45.49,
+    45.643, 46.935, 50.96,  53.365, 60.114, 67.621, 69.550,  69.645, 77.232,
+    83.559, 92.915, 95.673, 96.651, 99.211, 99.361, 101.355, 101.432
+  ];
+
+  for (let i = 0; i < wordSearchBoard.gameHeight; ++i) {
+    for (let j = 0; j < wordSearchBoard.gameWidth; ++j) {
+      let randomNum = Math.random() * 101.432;
+      wordSearchBoard.board[i][j].character = getCharater(frequency, randomNum);
+    }
+  }
+}
+
+/**
+ * Draws the game board onto the screen with the canvas api.
+ * @param {object} wordSearchBoard
+ */
+function drawGameBoard(wordSearchBoard) {
+  context.textAlign = 'center'
+  context.font = '30px Arial';
+  for (let i = 0; i < wordSearchBoard.gameHeight; ++i) {
+    for (let j = 0; j < wordSearchBoard.gameWidth; ++j) {
```

Comment:
Usually, when you need to iterate over a collection and you don't need indices for anything but getting an element - you can use for-of loop:

```js
for (const row of wordSearchBoard.board) {
  for (const square of row) {
    ...
   }
}
```

Reply:
Neat syntax, i did not know this existed. - Updated.